eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala

```diff
@@ -575,7 +575,7 @@ abstract class EventLog[A <: EventLogState](id: String) extends Actor with Event
         // 100% filtering coverage for certain replication network topologies.
         acc
       case (acc, e) =>
-        snr += 1L
+        snr = (snr + 1) max e.vectorTimestamp.value.getOrElse(id, 0)
```

Comment:
I think these adjustments should be made at the event of event recovery i.e. should be part of the disaster recovery code. It is not needed during normal event replication.